subprojects/core/src/test/groovy/org/gradle/api/internal/attributes/DefaultAttributesFactoryTest.groovy

```diff
@@ -180,8 +180,8 @@ class DefaultAttributesFactoryTest extends Specification implements TestsImmutab
 
     def "can detect incompatible attributes with different types when merging"() {
         given:
-        def set1 = factory.concat(factory.of(FOO, "foo1"), factory.of(OTHER_BAR, "bar1"))
-        def set2 = factory.concat(factory.of(FOO, "foo1"), factory.of(BAR, "bar2"))
+        def set1 = factory.concat(factory.of(FOO, "foo1"), factory.of(OTHER_BAR, "bar1")) // Object-typed bar
+        def set2 = factory.concat(factory.of(FOO, "foo1"), factory.of(BAR, "bar2")) // String-typed bar
```

Comment:
Do these still conflict if the value is the same?

Reply:
Yes.